### ISSUE TITLE
Conflict with other firebase plugins and google services

### ISSUE THREAD

**Issue by trey-rosius** [OneSignal/OneSignal-Flutter-SDK#47]
I've installed and ran onesignal on flutter, successfully. But when i added firebase auth and database plugins, i get the following build error. It's clear that there's some sort of conflict between the plugins.

Launching lib/main.dart on Android SDK built for x86 in debug mode...

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':app:debugCompileClasspath'.
> Could not find com.google.android.gms:play-services-measurement-base:11.8.0.
  Required by:
      project :app > project :firebase_core > com.google.firebase:firebase-core:11.8.0 > com.google.firebase:firebase-analytics:15.0.2
> Could not find com.google.android.gms:play-services-stats:11.8.0.
  Required by:
      project :app > project :firebase_core > com.google.firebase:firebase-core:11.8.0 > com.google.firebase:firebase-analytics:15.0.2
      project :app > project :onesignal > com.onesignal:OneSignal:3.10.3 > com.google.firebase:firebase-messaging:17.0.0 > com.google.firebase:firebase-iid:15.1.0
> Could not find com.google.android.gms:play-services-places-placereport:11.8.0.
  Required by:
      project :app > project :google_api_availability > com.google.android.gms:play-services-location:15.0.1
> Could not find com.google.firebase:firebase-measurement-connector:11.8.0.
  Required by:
      project :app > project :onesignal > com.onesignal:OneSignal:3.10.3 > com.google.firebase:firebase-messaging:17.0.0
> Could not find com.google.firebase:firebase-iid-interop:11.8.0.
  Required by:
      project :app > project :onesignal > com.onesignal:OneSignal:3.10.3 > com.google.firebase:firebase-messaging:17.0.0 > com.google.firebase:firebase-iid:15.1.0

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 11s
Gradle task assembleDebug failed with exit code 1
Exited (sigterm)

**Environment**
I'm using onesignal: ^1.0.0.
**heres my pubspec.yaml file***

> name: flutter-app
> description: A new Flutter project.
> 
> dependencies:
>   flutter:
>     sdk: flutter
>   flutter_localizations:
>     sdk: flutter
> 
>   
>   cupertino_icons: ^0.1.2
>   url_launcher: "^3.0.2"
> 
> dev_dependencies:
>   flutter_test:
>     sdk: flutter
> 
>   http: "^0.11.3+16"
>   shared_preferences: "^0.4.1"
> 
>   flutter_facebook_login: "^1.1.1"
>   firebase_auth: "^0.5.12"
>   google_sign_in: "^3.0.3"
>   flutter_webview_plugin: ^0.2.1+2
> 
>   firebase_database: ^1.0.4
>   intl_translation: "^0.17.0"
>   onesignal: ^1.0.0
>   
> 
>   image_picker: "^0.4.1"
>   intl: "^0.15.7"
>  
> 
>   geolocator: ^2.0.1
> 
> 
> flutter:
> 
>   uses-material-design: true
> 
> 
>   assets:
>     - assets/logo.png
>     - assets/logo_small.png
>     - assets/placeholder.jpg
> 
>  
>   fonts:
>     - family: lato
>       fonts:
>         - asset: assets/fonts/lato-regular.ttf
>         - asset: assets/fonts/lato-medium.ttf
>         - asset: assets/fonts/lato-bold.ttf
>   
>

**Comment by Nightsd01** [comment]
@trey-rosius Are you sure you followed our setup guide and [added the OneSignal Gradle Plugin](https://documentation.onesignal.com/v5.0/docs/flutter-sdk-setup#section-4-add-the-onesignal-gradle-plugin-android-) to your `app/build.gradle`?

**Comment by trey-rosius** [comment]
Yes I did,
I create two flutter applications. One With firebase Dependencies and another without firebase dependencies. 
The one with firebase dependencies can’t compile, while the other without firebase dependencies compiles and works.
I submitted an issue, it has the details
https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/47 <https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/47>
Best Regards
Rosius
